MapleServer2/Types/AdditionalEffects.cs

```diff
@@ -139,6 +147,69 @@ public AdditionalEffect AddEffect(AdditionalEffectParameters parameters)
 
         Parent.FieldManager.BroadcastPacket(BuffPacket.AddBuff(effect, Parent.ObjectId));
 
+        if (effect.LevelMetadata.SplashSkill != null)
+        {
+            for (int skill = 0; skill < effect.LevelMetadata.SplashSkill.Count; ++skill)
+            {
+                FireTriggerSkill(effect, effect.LevelMetadata.SplashSkill[skill], effect.ParentSkill, Parent, effect.Start);
+            }
+        }
+
+        EffectDotDamageMetadata dotDamage = effect.LevelMetadata.DotDamage;
+
+        if (dotDamage?.Rate != 0)
+        {
+            DamageSourceParameters dotParameters = new()
+            {
+                IsSkill = false,
+                GuaranteedCrit = false,
+                Element = (Element) dotDamage.Element,
+                RangeType = SkillRangeType.Special,
+                DamageType = (DamageType) dotDamage.DamageType,
+                DamageRate = dotDamage.Rate,
+            };
+
+            IFieldActor sourceActor = Parent.FieldManager.State.GetActor(parameters.Source);
+
+            // TODO: fix dot damage handling with damage numbers and game session tracking
+
+            DamageHandler damage = DamageHandler.CalculateDamage(dotParameters, sourceActor, Parent);
+            GameSession session = null;
+
+            if (sourceActor is Character character)
+            {
+                session = character.Value.Session;
+                Parent.Damage(damage, session);
+            }
+
+            List<DamageHandler> damages = new() { damage };
+            Parent.FieldManager.BroadcastPacket(SkillDamagePacket.Damage(parameters.ParentSkill, 0, Parent.Coord, Parent.Rotation, damages));
+
+            if (effect.Duration < effect.TickRate)
+            {
+                Task.Run(async () =>
+                {
+                    while (effect.IsAlive && Environment.TickCount < effect.End)
```

Comment:
Would Environment.TickCount - effect.Start < effect.Duration be a more stable approach?

Reply:
I think it would have the same problem, perhaps even worse.
But, the problem with tickCount is probably irrelevant in practice and not easy so solve, since its out only used tick in the server. I dont know how ticks are handled generally in game servers, when the tick reaches maximum size, because systems relying on tick are then bugged.

Reply:
Yeah, it's definitely an int overflow problem, but as long as it wraps around to the negatives I think the TickCount - start solution should provide the correct delta. The delta at the wrap (TickCount = -2147483646, Start = 2147483642) will be a large negative (-4294967288), but it'll be large enough that it gets wrapped around to to the correct value too (8).